jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TargetImageParameters.java

```diff
@@ -34,42 +36,52 @@
 
   private final AuthParameters auth;
 
-  @Nullable private String image;
-  private Set<String> tags = Collections.emptySet();
+  private Property<String> image;
+  private SetProperty<String> tags;
   @Nullable private String credHelper;
 
   @Inject
   public TargetImageParameters(ObjectFactory objectFactory) {
     auth = objectFactory.newInstance(AuthParameters.class, "to.auth");
+    image = objectFactory.property(String.class);
+    tags = objectFactory.setProperty(String.class).empty();
   }
 
   @Input
   @Nullable
-  @Optional
   public String getImage() {
     if (System.getProperty(PropertyNames.TO_IMAGE) != null) {
       return System.getProperty(PropertyNames.TO_IMAGE);
     }
-    return image;
+    return image.getOrNull();
   }
 
   public void setImage(String image) {
-    this.image = image;
+    this.image.set(image);
+  }
+
+  public void setImage(Provider<String> image) {
+    if (image != null) {
+      this.image.set(image);
+    }
   }
 
   @Input
-  @Optional
```

Comment:
I think this should be preserved. The user doesn't have to configure this parameter.